src/main/scala/com/amazon/deequ/analyzers/runners/MetricCalculationException.scala

```diff
@@ -21,12 +21,12 @@ abstract class MetricCalculationException(message: String) extends Exception(mes
 class MetricCalculationRuntimeException(message: String)
   extends MetricCalculationException(message) {
 
-  def this(message: String, cause: Throwable) {
+  def this(message: String, cause: Exception) {
     this(message)
     initCause(cause)
   }
 
-  def this(cause: Throwable) {
+  def this(cause: Exception) {
     this(Option(cause).map(_.toString).orNull, cause)
```

Comment:
We should have better logic for the case where cause can be null. We should not pass the nulls to the constructor.

Reply:
When are we expecting the cause to be null?

Reply:
Yes, I was wondering why we are explicitly handling the case when `cause` can be null. Are we expecting it to be null at any point?

Reply:
No, not really. We could check it and invoke the constructor with a custom runtime exception to be super-duper safe, but we should not propagate nulls.